src/sidebar/util/version-data.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+/**
+ * An object representing version, browser and user data
+ *
+ * @typedef {Object} VersionData
+ * @property {string} timestamp - Current Date as string
+ * @property {string} url - Current document URL or 'N/A' if no available URL
+ * @property {string} fingerprint - document PDF fingerprint or 'N/A'
+ * @property {string} account - user id; with display name if available or 'N/A'
+ *                              if not logged in
+ * @property {string} userAgent - current browser user agent
+ * @property {string} version - current client app version
+ */
+
+/**
+ * An object representing user info
+ *
+ * @typedef {Object} UserInfo
+ * @property {string=} userid
+ * @property {string=} displayName
+ */
+
+/**
+ * An object representing document metadata
+ *
+ * @typedef {Object} DocMetadata
+ * @property {string=} documentFingerprint - optional PDF fingerprint for
+ *                     current document
+ */
+
+/**
+ * An object representing document info
+ *
+ * @typedef {Object} DocumentInfo
+ * @property {string=} url - current document URL
+ * @property {DocMetadata} metadata - document metadata
+ */
+
+class VersionData {
+  /**
+   * @param {UserInfo} userInfo
+   * @param {DocumentInfo} documentInfo
+   * @param {Object} window_ - test seam
+   * @return {VersionData}
+   */
+  constructor(userInfo, documentInfo, window_ = window) {
+    const noValueString = 'N/A';
+    const docMeta = documentInfo.metadata;
+
+    let accountString = noValueString;
+    if (userInfo.userid) {
+      accountString = userInfo.userid;
+      if (userInfo.displayName) {
+        accountString = `${userInfo.displayName} (${accountString})`;
+      }
+    }
+
+    this.timestamp = new Date().toString();
+    this.url = documentInfo.uri || noValueString;
+    this.fingerprint =
+      docMeta && docMeta.documentFingerprint
+        ? docMeta.documentFingerprint
+        : noValueString;
+    this.account = accountString;
+    this.userAgent = window_.navigator.userAgent;
+    this.version = '__VERSION__';
+  }
+
+  /**
+   * Return a single, encoded URL string of version data suitable for use in
+   * a querystring (as the value of a single parameter)
+   *
+   * @return {String} - URI-encoded string
```

Comment:
For primitive values, `string`/`number`/`boolean` is preferred to `String`/`Number`/`Boolean`. Both mean the same thing as far as I am aware, but that's just the convention we've used.